src/OVAL/oval_probe.c

```diff
@@ -175,14 +175,7 @@ static void __init_once(void)
 
 const char *oval_subtype_to_str(oval_subtype_t subtype)
 {
-        oval_subtypedsc_t *d;
-
-        __init_once();
-
-        d = oscap_bfind(OSCAP_GSYM(__s2n_tbl), __s2n_tbl_count, sizeof(oval_subtypedsc_t), &subtype,
-                        (int(*)(void *, void *))__s2n_tbl_cmp);
-
-        return (d == NULL ? NULL : d->name);
+	return oval_subtype_get_text(subtype);
```

Comment:
Maybe you should add something like this
```
return oval_subtype_is_valid(subtype) ? oval_subtype_get_text(subtype) : NULL;
```
because currently somebody who use public API can rely on `NULL value`.